.github/workflows/gitlab-push.yml

```diff
@@ -16,6 +16,37 @@ env:
   PR_NUMBER: ${{ github.event.pull_request.number }}
 
 jobs:
+  push_to_gitlab_improved:
+    name: Push To GitLab (Improved)
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Push to GitLab
+        run: |
+          set -eux
+          URL="https://push-from-github:${{ secrets.PUSH_TO_GITLAB_TOKEN }}@gitlab.com/dfinity-lab/public/ic.git"
+
+          if git remote get-url gitlab >/dev/null 2>&1; then
```

Comment:
I'm guessing `git remote add` fails if the remote already exists? in which case would the `gitlab` remote exist already?